pkg/controller/worker/machines.go

```diff
@@ -205,3 +219,15 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 
 	return nil
 }
+
+func (w *workerDelegate) generateWorkerPoolHash(pool extensionsv1alpha1.WorkerPool, serverGroupDependency *api.ServerGroupDependency) (string, error) {
+	var additionalHashData = []string{}
+
+	// Include the given worker pool dependencies into the hash.
+	if serverGroupDependency != nil {
+		additionalHashData = append(additionalHashData, serverGroupDependency.Name, serverGroupDependency.ID)
```

Comment:
Out of curiosity: Wouldn't it be sufficient to just add the servergroup dependency `id` to the hash? I mean this one should be unique.